libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/infopanel/InfoPanelHeaderComponent.kt

```diff
@@ -39,22 +44,54 @@ internal class InfoPanelHeaderComponent(
             binding.arrivedText.isVisible = it == NavigationState.Arrival
         }
 
-        destinationState.observe {
+        destinationViewModel.state.observe {
             val placeName = it.destination?.features?.firstOrNull()?.placeName()
-            binding.poiName.text =
-                placeName ?: resources.getString(R.string.mapbox_drop_in_dropped_pin)
+            binding.poiName.text = placeName
+                ?: resources.getString(R.string.mapbox_drop_in_dropped_pin)
         }
 
         binding.routePreview.setOnClickListener {
-            dispatch(RoutesAction.FetchAndSetRoute)
+            updateNavigationStateWhenRouteIsReady(NavigationState.RoutePreview)
         }
 
         binding.startNavigation.setOnClickListener {
-            dispatch(RoutesAction.StartNavigation)
+            updateNavigationStateWhenRouteIsReady(NavigationState.ActiveNavigation)
         }
 
         binding.endNavigation.setOnClickListener {
-            dispatch(RoutesAction.StopNavigation)
+            routesViewModel.invoke(RoutesAction.SetRoutes(emptyList()))
+            destinationViewModel.invoke(DestinationAction.SetDestination(null))
+            navigationStateViewModel.invoke(
+                NavigationStateAction.Update(NavigationState.FreeDrive)
+            )
         }
     }
+
+    private fun updateNavigationStateWhenRouteIsReady(navigationState: NavigationState) {
+        val lastPoint = locationViewModel.lastPoint
+            ?: error("Cannot fetch routes when current location is unknown")
+        val destination = destinationViewModel.state.value.destination
+            ?: error("Cannot fetch routes when destination is unknown")
```

Comment:
Same as above

Reply:
Moving this conversation to this thread https://github.com/mapbox/mapbox-navigation-android/pull/5594#discussion_r833730652